microprofile/config-smallrye/src/main/java/org/wildfly/extension/microprofile/config/smallrye/deployment/SubsystemDeploymentProcessor.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.microprofile.config.smallrye.deployment;
+
+import java.util.List;
+
+import io.smallrye.config.SmallRyeConfigBuilder;
+import io.smallrye.config.inject.ConfigExtension;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+import org.jboss.as.ee.weld.WeldDeploymentMarker;
+import org.jboss.as.server.deployment.AttachmentKey;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.Phase;
+import org.jboss.as.weld.deployment.WeldPortableExtensions;
+import org.jboss.modules.Module;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceRegistry;
+import org.wildfly.extension.microprofile.config.smallrye.ServiceNames;
+
+/**
+ */
+public class SubsystemDeploymentProcessor implements DeploymentUnitProcessor {
+
+    public static final AttachmentKey<Config> CONFIG = AttachmentKey.create(Config.class);
+    public static final AttachmentKey<ConfigProviderResolver> CONFIG_PROVIDER_RESOLVER = AttachmentKey.create(ConfigProviderResolver.class);
+
+    /**
+     * See {@link Phase} for a description of the different phases
+     */
+    public static final Phase PHASE = Phase.POST_MODULE;
+
+    /**
+     * The relative order of this processor within the {@link #PHASE}.
+     * The current number is large enough for it to happen after all
+     * the standard deployment unit processors that come with JBoss AS.
+     */
+    public static final int PRIORITY = 0x4000;
```

Comment:
Are we no longer putting these in the `Phase`?

Reply:
@jamezp Good catch, I set it like that when the code was in wildfly-extras repo and was not "officially" in WildFly code base.

Is it acceptable to use existing priority + offset (e.g. `Phase.POST_MODULE_WELD_PORTABLE_EXTENSIONS + 10`) ?